Bundled Kokkos + CUDA

It looks like we can't configure our bundled copy of Kokkos to use CUDA - is this expected? I see the following message:

```
-- DEAL_II_WITH_KOKKOS has unmet external dependencies.
-- 
-- DEAL_II_WITH_KOKKOS successfully set up with bundled packages.
-- 
-- Include /home/drwells/Documents/Code/CPP/dealii/cmake/configure/configure_40_cuda.cmake
-- DEAL_II_WITH_CUDA has unmet external dependencies.
CMake Error at cmake/configure/configure_40_cuda.cmake:103 (message):
  

  Could not find any suitable cuda library!

  deal.II can only be compiled with Cuda support if Kokkos was built with
  Cuda support!
```

How hard would it be to build our own Kokkos with CUDA support? I would like to do this so I can test step-64.

--- a/cmake/configure/configure_40_cuda.cmake
+++ b/cmake/configure/configure_40_cuda.cmake
@@ -23,7 +23,13 @@
 set(DEAL_II_WITH_CUDA FALSE CACHE BOOL "")
 
 macro(feature_cuda_find_external var)
-  if(NOT Kokkos_ENABLE_CUDA)
+  if(DEAL_II_FEATURE_KOKKOS_BUNDLED_CONFIGURED)
+    set(CUDA_ADDITIONAL_ERROR_STRING
+      ${CUDA_ADDITIONAL_ERROR_STRING}
+      "deal.II's bundled version of Kokkos only supports the Serial backend and therefore cannot be used with Cuda."
+      )
+    set(${var} FALSE)
+  elseif(NOT Kokkos_ENABLE_CUDA)
     set(CUDA_ADDITIONAL_ERROR_STRING
       ${CUDA_ADDITIONAL_ERROR_STRING}
       "deal.II can only be compiled with Cuda support if Kokkos was built with Cuda support!"
@@ -113,7 +119,7 @@ macro(feature_cuda_error_message)
   message(FATAL_ERROR "\n"
     "Could not find any suitable cuda library!\n"
     ${CUDA_ADDITIONAL_ERROR_STRING}
-    "\nPlease ensure that a cuda library is installed on your computer\n"
+    "\nPlease ensure that a cuda library is installed on your computer and deal.II is configured to use an external Kokkos installation.\n"
     )
 endmacro()
 

